python/lsst/ts/externalscripts/base_make_calibrations.py

```diff
@@ -599,6 +605,186 @@ async def verify_calib(self, image_type, job_id_calib, exposure_ids_dict):
+
+        job_idverify : `str`
+            Job ID returned by OCPS during previous calibration
+            verification pipetask call.
+
+        Returns
+        -------
+        certify_calib : `bool`
+            Booolean indicating if the calibration should be certified or not.
+
+        numStatErrors : `dict`[`str`][`str`]
+            Dictionary with the total number of tests failed per exposure and
+            per cp_verify test type. If there are not any tests that failed,
+            `None` will be returned.
+
+        thresholds : `dict`[`str`][`int`]
+            Dictionary reporting the different thresholds used to decide
+            whether a calibration should be certified or not (see `Notes`
+            below). If there are not any tests that failed,
+            `None` will be returned.
+
+        verify_stats : `dict`
+            Statistics from cp_verify.
+        """
+        # Collection name containing the verification outputs.
+        verifyCollection = f"u/ocps/{job_id_verify}"
+        # Collection that the calibration was constructed in.
+        genCollection = f"u/ocps/{job_id_calib}"
+        if image_type == "BIAS":
+            verify_stats_string = "verifyBiasStats"
+        elif image_type == "DARK":
+            verify_stats_string = "verifyDarkStats"
+        elif image_type == "FLAT":
+            verify_stats_string = "verifyFlatStats"
+        else:
+            raise RuntimeError(
+                f"Verification is not currently implemented for {image_type}"
+            )
+        butler = dafButler.Butler(
+            self.config.repo, collections=[verifyCollection, genCollection]
+        )
+        # verify_stats is a dictionary with the verification
+        # tests that failed, if any. See `cp_verify`.
+        verify_stats = butler.get(verify_stats_string, instrument=self.instrument_name)
+
+        if "FAILURES" in verify_stats:
+            (
+                certify_calib,
+                numStatErrors,
+                failure_thresholds,
+            ) = await self.count_failed_verification_tests(verify_stats)
+
+            return certify_calib, numStatErrors, failure_thresholds, verify_stats
+        else:
+            # Nothing failed
+            return True, None, None, verify_stats
+
+    async def count_failed_verification_tests(self, verify_stats):
+        """Count number of tests that failed cp_verify.
+
+        Parameters
+        ----------
+        verify_stats : `dict`
+            Statistics from cp_verify.
+
+        Returns
+        -------
+        certify_calib : `bool`
+            Boolean assessing whether the calibration should be certified.
+
+        total_counter_failed_tests : `dict`[`str`][`str`]
+            Dictionary with the total number of tests failed per exposure and
+            per cp_verify test type. If there are not any tests that failed,
+            `None` will be returned.
+
+        thresholds : `dict`[`str`][`int`]
```

Comment:
Is this one supposed to be None if it fails? I see there is a return of 2 values if it succeeds and 3 values if it fails. I think that will be problematic to unpack.